tests/components/tuya/test_climate.py

```diff
@@ -2,22 +2,99 @@
 
 from __future__ import annotations
 
+from typing import cast
 from unittest.mock import patch
 
 import pytest
 from syrupy.assertion import SnapshotAssertion
-from tuya_sharing import CustomerDevice
+from tuya_sharing import CustomerDevice, Manager
 
+from homeassistant.components.climate import HVACMode
 from homeassistant.components.tuya import ManagerCompat
-from homeassistant.const import Platform
+from homeassistant.components.tuya.climate import (
+    TuyaClimateEntity,
+    TuyaClimateEntityDescription,
+)
+from homeassistant.components.tuya.const import DPCode
+from homeassistant.const import Platform, UnitOfTemperature
 from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import HomeAssistantError
 from homeassistant.helpers import entity_registry as er
 
 from . import DEVICE_MOCKS, initialize_entry
 
 from tests.common import MockConfigEntry, snapshot_platform
 
 
+class DummyDevice:
+    """Dummy device for testing."""
+
+    def __init__(self, function, status) -> None:
+        """Initialize the dummy device."""
+        self.function = function
+        self.status = status
+        self.id = "dummy"
+        self.name = "Dummy"
+        self.product_name = "Dummy"
+        self.product_id = "dummy"
+        self.status_range = {}
+        self.online = True
+
+
+class DummyManager:
+    """Dummy manager for testing."""
+
+    def send_commands(self, device_id: str, commands: list) -> None:
+        """Send commands to the device."""
+
+
+class DummyFunction:
+    """Dummy function for testing."""
+
+    def __init__(self, type_: str, values: str) -> None:
+        """Initialize the dummy function."""
+        self.type = type_
+        self.values = values
+
+
+def make_climate_entity(function, status):
+    """Make a dummy climate entity for testing."""
+    return TuyaClimateEntity(
+        cast("CustomerDevice", DummyDevice(function, status)),
+        cast("Manager", DummyManager()),
+        TuyaClimateEntityDescription(key="kt", switch_only_hvac_mode=HVACMode.COOL),
+        UnitOfTemperature.CELSIUS,
+    )
+
+
+def test_fan_mode_windspeed() -> None:
+    """Test fan mode with windspeed."""
+    entity = make_climate_entity(
+        {"windspeed": DummyFunction("Enum", '{"range": ["1", "2"]}')},
+        {"windspeed": "2"},
+    )
+    assert entity.fan_mode == "2"
+    entity.set_fan_mode("1")
```

Comment:
`set_fan_mode` should be tested using service calls - not direct function calls on the entity